app/modules/tcs34725.c

```diff
@@ -0,0 +1,371 @@
+	if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
+		return luaL_error(L, "wrong arg range");
+	}
+	sda = luaL_checkinteger(L, 1);
+	scl = luaL_checkinteger(L, 2);
+	platform_i2c_setup(TCS34725_BUS_ID, sda, scl, PLATFORM_I2C_SPEED_SLOW);
+	
+	/* Make sure we have the right IC (0x44 = TCS34725 and TCS34721) */
+	id = tcs34725Read8(TCS34725_ID);
+	dbg_printf("id: %x\n",id);
+	if (id != 0x44) {
+		luaL_error(L, "No TCS34725 found.");
+		return 0;
+	}
+
+	/* Enable the device */
+	// TODO this doesn't quite work as it should. Enable is non-blocking.
+	if (tcs34725Enable(L) != 0) {
+		luaL_error(L, "TCS34725 Not initialising.");
+		return 0;
+	}
+	
+	// TODO Block here and wait for _tcs34725Initialised?
+
+	/* This needs to take place after the initialisation flag! */
+	// tcs34725SetIntegrationTime(TCS34725_INTEGRATIONTIME_2_4MS, L);
+	// tcs34725SetGain(TCS34725_GAIN_60X, L);
+	lua_pushinteger(L, 1);
+	return 1;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Sets the integration time to the specified value
+*/
+/**************************************************************************/
+uint8_t tcs34725LuaSetIntegrationTime(lua_State* L)
+{
+	tcs34725IntegrationTime_t it = luaL_checkinteger(L, 1);
+	return tcs34725SetIntegrationTime(it,L);
+}
+
+/**************************************************************************/
+/*!
+		@brief	Sets the integration time to the specified value
+*/
+/**************************************************************************/
+uint8_t tcs34725SetIntegrationTime(tcs34725IntegrationTime_t it, lua_State* L)
+{
+	if (!_tcs34725Initialised)
+	{
+		tcs34725Init(L);
+	}
+
+	tcs34725Write8(TCS34725_ATIME, it);
+	_tcs34725IntegrationTime = it;
+
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Sets gain to the specified value from Lua
+*/
+/**************************************************************************/
+uint8_t tcs34725LuaSetGain(lua_State* L)
+{
+	tcs34725Gain_t gain = luaL_checkinteger(L, 1);
+	return tcs34725SetGain(gain,L);
+}
+
+/**************************************************************************/
+/*!
+		@brief	Sets gain to the specified value
+*/
+/**************************************************************************/
+uint8_t tcs34725SetGain(tcs34725Gain_t gain, lua_State* L)
+{
+  if (!_tcs34725Initialised)
+  {
+    tcs34725Init(L);
+  }
+
+  tcs34725Write8(TCS34725_CONTROL, gain);
+  _tcs34725Gain = gain;
+
+  return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Reads the raw red, green, blue and clear channel values
+*/
+/**************************************************************************/
+uint8_t tcs34725GetRawData(lua_State* L)
+{
+	uint16_t r;
+	uint16_t g;
+	uint16_t b;
+	uint16_t c;
+	
+	if (!_tcs34725Initialised)
+	{
+		tcs34725Init(L);
+	}
+
+	/* ToDo: Insert a blocky delay until the data is ready! */
```

Comment:
How long would the delay be?
Can the delay be implemented on Lua level instead here?